pkg/catalogv2/git/download.go

```diff
@@ -20,103 +10,110 @@ const (
+func (er *extendedRepo) CloneHead(branch string) (string, error) {
+	err := er.cloneOrOpen(branch)
+	if err != nil {
+		return "", err
+	}
+
+	err = er.hardReset("HEAD", "")
+	if err != nil {
+		return "", err
+	}
+
+	return er.getCurrentCommit()
 }
 
+// Update builds the configuration to update an existing repository
 func Update(secret *corev1.Secret, namespace, name, gitURL, branch string, insecureSkipTLS bool, caBundle []byte) (string, error) {
 	git, err := gitForRepo(secret, namespace, name, gitURL, insecureSkipTLS, caBundle)
 	if err != nil {
 		return "", err
 	}
 
-	if isBundled(git) && settings.SystemCatalog.Get() == "bundled" {
+	if isBundled(git.GetConfig()) && settings.SystemCatalog.Get() == "bundled" {
 		return Head(secret, namespace, name, gitURL, branch, insecureSkipTLS, caBundle)
 	}
 
-	commit, err := git.Update(branch)
-	if err != nil && isBundled(git) {
+	commit, err := git.UpdateToLatestRef(branch)
+
+	if err != nil && isBundled(git.GetConfig()) {
 		return Head(secret, namespace, name, gitURL, branch, insecureSkipTLS, caBundle)
 	}
+
 	return commit, err
 }
 
-func Ensure(secret *corev1.Secret, namespace, name, gitURL, commit string, insecureSkipTLS bool, caBundle []byte) error {
-	if commit == "" {
-		return nil
-	}
-	git, err := gitForRepo(secret, namespace, name, gitURL, insecureSkipTLS, caBundle)
+// UpdateToLatestRef will check if repository exists, if exists will check for latest commit and update to it.
+// If the repository does not exist will try cloning again.
+func (er *extendedRepo) UpdateToLatestRef(branch string) (string, error) {
+	err := er.cloneOrOpen(branch)
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	return git.Ensure(commit)
-}
-
-func isBundled(git *git) bool {
-	return strings.HasPrefix(git.Directory, staticDir)
-}
-
-func gitForRepo(secret *corev1.Secret, namespace, name, gitURL string, insecureSkipTLS bool, caBundle []byte) (*git, error) {
-	isGitSSH, err := isGitSSH(gitURL)
+	err = er.hardReset("HEAD", "")
 	if err != nil {
-		return nil, fmt.Errorf("failed to verify the type of URL %s: %w", gitURL, err)
-	}
-	if !isGitSSH {
-		u, err := url.Parse(gitURL)
-		if err != nil {
-			return nil, fmt.Errorf("failed to parse URL %s: %w", gitURL, err)
-		}
-		if u.Scheme != "http" && u.Scheme != "https" {
-			return nil, fmt.Errorf("invalid git URL scheme %s, only http(s) and git supported", u.Scheme)
-		}
+		return "", err
 	}
-	dir := gitDir(namespace, name, gitURL)
-	headers := map[string]string{}
-	if settings.InstallUUID.Get() != "" {
-		headers["X-Install-Uuid"] = settings.InstallUUID.Get()
+
+	commit, err := er.getCurrentCommit()
+	if err != nil {
+		return commit, err
 	}
-	// convert caBundle to PEM format because git requires correct line breaks, header and footer.
-	if len(caBundle) > 0 {
-		caBundle = convertDERToPEM(caBundle)
-		insecureSkipTLS = false
+	// check if the currentCommit == latest commit in remote repo (not yet implemented by go-git pacakge)
```

Comment:
As per the other comment, `go-git` provides ls-remote.
```suggestion
	// check if the currentCommit == latest commit in remote repo (not yet implemented by go-git package)
```